### PR TITLE
[REF] UnitConverter.h

### PR DESCRIPTION
**Pull request by SamuilsRulovs**
- Replaced std::shared_ptr with EXPRESSReference
- included "EXPRESS/EXPRESSReference.h" library
- Added InconsistentModellingExceptions

### CODE REVIEW

File: Core/src/IfcGeometryConverter/UnitConverter.h [hunk 1]
```diff
@@ -72,8 +73,11 @@ namespace OpenInfraPlatform
 				*/
 				double getFactorFor(const typename IfcEntityTypesT::IfcUnitEnum& unit_type) const
 				{
-					if (!m_unit_assignment)
-						return 1.;
+					
+					if (!m_unit_assignment) {
+						throw oip::InconsistentModellingException(m_unit_assignment, "m_unit_assignmentt is not defined");
+					}
+
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/UnitConverter.h]: Leave it as it was. This exception is already handled in `setIfcProject`.

File: Core/src/IfcGeometryConverter/UnitConverter.h [hunk 2]
```diff
@@ -121,8 +125,9 @@ namespace OpenInfraPlatform
 				*/
 				double getFactorFor(const typename IfcEntityTypesT::IfcDerivedUnitEnum& unit_type) const
 				{
-					if (!m_unit_assignment)
-						return 1.;
+					if (!m_unit_assignment) {
+						throw oip::InconsistentModellingException(m_unit_assignment, "m_unit_assignmentt is not defined");
+					}
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/UnitConverter.h]: Similar to above.

**Review by pjanck — CHANGES_REQUESTED** [review]
There are some `.lock()` function calls in the file. Since you've changed the functions to now take `EXPRESSReference` instead of `std::shared_ptr`, these can be almost everywhere removed.

File: Core/src/IfcGeometryConverter/UnitConverter.h (hunk 1 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/UnitConverter.h]: Resolved: e854d00

File: Core/src/IfcGeometryConverter/UnitConverter.h (hunk 2 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/UnitConverter.h]: Resolved: e854d00

**Review by pjanck — APPROVED** [review]

**Pull request merged**